src/coreclr/jit/gentree.cpp

```diff
@@ -18198,6 +18206,59 @@ GenTree* Compiler::gtNewSimdBinOpNode(genTreeOps  op,
             break;
         }
 
+        case GT_LSH:
+        case GT_RSH:
+        case GT_RSZ:
+        {
+            assert(!varTypeIsByte(simdBaseType) && !varTypeIsFloating(simdBaseType));
+            assert((op != GT_RSH) || !varTypeIsUnsigned(simdBaseType));
+
+            // "over shifting" is platform specific behavior. We will match the C# behavior
+            unsigned shiftCountMask = (genTypeSize(simdBaseType) * 8) - 1;
+
+            if (op2->IsCnsIntOrI())
+            {
+                op2->AsIntCon()->gtIconVal &= shiftCountMask;
+            }
+            else
+            {
+                op2 = gtNewOperNode(GT_AND, simdBaseType, op2, gtNewIconNode(shiftCountMask));
```

Comment:
`genActualType(simdBaseType)`?

If the `AND` can be `TYP_LONG`, `gtNewIconNode` needs to be passed the right type too.

Reply:
this is `TYP_INT` and has been fixed.